### ISSUE TITLE
Scan QR code - should prompt for password after scanning successfully?

### ISSUE THREAD

**Issue by Qrysto** [NemProject/NEMAndroidApp#237]
When I scan QR codes to import accounts, it always asks me for password immediately, then after I enter my password, it shows the result: "Failed to import account". It's annoying when I have to scan QR codes  again and again, the process is slow. Is there any problem if we check the QR code validity before prompting user for password? Is password only needed when the QR code is valid and it's really going to be added to the wallet?

Version: 1.0.51.b3315 BETA

Main net address: NBPL6F-J2OQGS-P6IJWZ-PMSL67-BQ6U3I-LIHVCA-DUO5

**Comment by am4n0w4r** [comment]
This is a security feature. QR code has encrypted private key so no one can steal all your money just by e.g. taking photo of your QR or hacking your email with code somewhere in attachments.
If you're seeing password prompt, it already means that QR code is valid and you have to enter password _with which it was created_, this doesn't always and usually shouldn't be a wallet password - just for a note.
If you see that message after entering password - it means that password was wrong, decryption is not possible and you can't import that account.
Please make sure you're entering the right password first, this system haven't been changed for a while and worked fine.

**Comment by jabo38** [comment]
So the issue is when a person enters a wrong password, they don't know exactly what they did wrong.  The QR could have been bad, it could have been a problem with the app, or it could have been a wrong password.  The user won't know the difference with a generic "failed to import account".  

Ideally, we need an error message that states why it didn't import; in this case the best message is "incorrect password" which is pretty standard on most apps and is what a user will be expecting.

**Comment by Qrysto** [comment]
I don't think that I entered the wrong password. I've done it dozens of times, cleared all data to start again from scratch (create a new password), then entered it many times again (still failed), then repeat.... Maybe it's some error in importing the account, I don't know but the message should be clearer

**Comment by jabo38** [comment]
Bounty paid. Tx hash. 41fa867bd8c6fb2ee52a82c913f020c53faa9cdb98cd3083b8ffe16e0fa735dc

**Comment by Qrysto** [comment]
thanks, received in my wallet :smile:

**Comment by am4n0w4r** [comment]
Interesting issue...
@Qrysto can I see QRs that you can't import, with passwords to import, and account details?
I can give you my email if you don't want to share this publicly.
Also I need to know where QRs came from and were they created with default password checkbox set, or with custom password when exporting?
@jabo38 ok, I'll change the message to "Incorrect password".

**Comment by Qrysto** [comment]
@am4n0w4r ok i'll try to give you more info tonight when i'm home

**Comment by Qrysto** [comment]
Ohhhh f*** I've figured it out lol
Sorry for forgetting this a few days now. Turns out that the password box appears after the QR is scanned is the *import password* for the QRs, not *my password* that I've set when I open NEM app the first time!
Now I've noticed, the password box after QRs are scanned is labeled "Enter import password" while the other one is just "ENTER PASSWORD". It's hard to know these were different. Probably they should be differentiated more so that we can know they're different.

**Comment by am4n0w4r** [comment]
Ok, I see :) Actually I mentioned it in my message :)

**Comment by am4n0w4r** [comment]
Changed message to "Importing failed. Incorrect password". For now this is the only thing to resolve here I see.
